docker-versions-maven-plugin/src/main/java/com/github/cafapi/docker_versions/plugins/IgnoreVersionsHelper.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2024 Open Text.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.cafapi.docker_versions.plugins;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Helper class for managing ignore versions.
+ */
+public final class IgnoreVersionsHelper
+{
+    private static final List<String> DEFAULT_IGNORE_VERSIONS = Arrays.asList(new String[]{"latest", "stable"});
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(IgnoreVersionsHelper.class);
+
+    public static List<String> getRelevantTags(
+        final Set<IgnoreVersion> configuredIgnoreVersions,
+        final List<String> tags,
+        final String repository)
+    {
+        LOGGER.debug("Configured ignore versions: {}", configuredIgnoreVersions);
+        final Map<String, Set<IgnoreVersion>> imageIgnoreVersionsLookup = new HashMap<>();
+        return tags.stream()
+            .filter(t -> !isIgnoredVersion(configuredIgnoreVersions, imageIgnoreVersionsLookup, repository, t))
```

Comment:
It is not good practice to update a map inside a filter like this.  It makes presumptions about how the filter will be used, 
 This may need to be made into more standard procedural code or done a slightly different way.